File: python/meteroid/models/plan_event_data.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .plan_id import PlanId
from .plan_status_enum import PlanStatusEnum
from .plan_type_enum import PlanTypeEnum


@dataclasses.dataclass
class PlanEventData(BaseModel):
    created_at: datetime

    currency: str

    name: str

    plan_id: PlanId

    plan_type: PlanTypeEnum

    status: PlanStatusEnum

    version: int

    description: t.Optional[str] = None

File: go/plan_event_data.go
// this file is @generated
package meteroid

import "time"

type PlanEventData struct {
	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	Currency string `json:"currency"`

	Description *string `json:"description,omitempty"`

	Name string `json:"name"`

	PlanId PlanId `json:"plan_id"`

	PlanType PlanTypeEnum `json:"plan_type"`

	Status PlanStatusEnum `json:"status"`

	Version int32 `json:"version"`
}
